src/instance.rs

```diff
@@ -1016,14 +1016,30 @@ impl<T> Py<T> {
             .setattr(attr_name, value.into_py(py).into_bound(py))
     }
 
+    /// Deprecated form of [`call_bound`][Py::call_bound].
+    #[cfg_attr(
+        not(feature = "gil-refs"),
+        deprecated(
+            since = "0.21.0",
+            note = "`call` will be replaced by `call_bound` in a future PyO3 version"
+        )
+    )]
+    #[inline]
+    pub fn call<A>(&self, py: Python<'_>, args: A, kwargs: Option<&PyDict>) -> PyResult<PyObject>
+    where
+        A: IntoPy<Py<PyTuple>>,
+    {
+        self.call_bound(py, args, kwargs.map(PyDict::as_borrowed).as_deref())
+    }
+
     /// Calls the object.
     ///
     /// This is equivalent to the Python expression `self(*args, **kwargs)`.
-    pub fn call(
+    pub fn call_bound(
         &self,
         py: Python<'_>,
         args: impl IntoPy<Py<PyTuple>>,
-        kwargs: Option<&PyDict>,
+        kwargs: Option<&Bound<'_, PyDict>>,
```

Comment:
It's a reasonable question. In other APIs e.g. `PyIterator::from_bound_object` we just added in #3702 I've been choosing to use `&Bound<'py, PyAny>` because there's been no need to transfer ownership, and `Option<&Bound<'_, PyDict>>` seemed to be the logical parallel. Here we also don't need to transfer ownership of a python reference, so taking `&Bound<'_, PyDict>` avoids a needless reference count increase / decrease cycle.

If we're concerned by the double pointer indirection we _could_ take `Option<Borrowed<'_, '_, PyDict>>` instead, though so far I've tried to avoid making the user to deal with `Borrowed` as anything other than a return value from specific APIs where returning a `Bound` would force the reference count increase.